diff --git a/fallback.c b/fallback.c
index 9523318c..d83b5080 100644
--- a/fallback.c
+++ b/fallback.c
@@ -8,7 +8,7 @@ char *rcs_fallback="$Id: fallback.c,v 1.24 1996/04/22 18:00:37 roberto Exp rober
 #include <stdio.h>
 #include <string.h>
  
-#include "mem.h"
+#include "luamem.h"
 #include "fallback.h"
 #include "opcode.h"
 #include "lua.h"
diff --git a/func.c b/func.c
index 0e15c2fc..3ac3ccf7 100644
--- a/func.c
+++ b/func.c
@@ -2,7 +2,7 @@
 
 #include "luadebug.h"
 #include "table.h"
-#include "mem.h"
+#include "luamem.h"
 #include "func.h"
 #include "opcode.h"
 
diff --git a/hash.c b/hash.c
index 90d81293..2dcbc72c 100644
--- a/hash.c
+++ b/hash.c
@@ -6,7 +6,7 @@
 char *rcs_hash="$Id: hash.c,v 2.31 1996/07/12 20:00:26 roberto Exp roberto $";
 
 
-#include "mem.h"
+#include "luamem.h"
 #include "opcode.h"
 #include "hash.h"
 #include "table.h"
diff --git a/inout.c b/inout.c
index 0f79b9c7..aa656a23 100644
--- a/inout.c
+++ b/inout.c
@@ -16,7 +16,7 @@ char *rcs_inout="$Id: inout.c,v 2.42 1996/09/24 21:46:44 roberto Exp roberto $";
 #include "table.h"
 #include "tree.h"
 #include "lua.h"
-#include "mem.h"
+#include "luamem.h"
 
 
 /* Exported variables */
diff --git a/lex.c b/lex.c
index b4513919..27d9d036 100644
--- a/lex.c
+++ b/lex.c
@@ -4,7 +4,7 @@ char *rcs_lex = "$Id: lex.c,v 2.38 1996/11/08 12:49:35 roberto Exp roberto $";
 #include <ctype.h>
 #include <string.h>
 
-#include "mem.h"
+#include "luamem.h"
 #include "tree.h"
 #include "table.h"
 #include "lex.h"
diff --git a/lua.stx b/lua.stx
index a90f5690..98739461 100644
--- a/lua.stx
+++ b/lua.stx
@@ -7,7 +7,7 @@ char *rcs_luastx = "$Id: lua.stx,v 3.40 1996/09/25 21:52:00 roberto Exp roberto
 #include <string.h>
 
 #include "luadebug.h"
-#include "mem.h"
+#include "luamem.h"
 #include "lex.h"
 #include "opcode.h"
 #include "hash.h"
diff --git a/luamem.c b/luamem.c
index db67fb87..59ef2ec7 100644
--- a/luamem.c
+++ b/luamem.c
@@ -1,13 +1,13 @@
 /*
-** mem.c
+** luamem.c
 ** TecCGraf - PUC-Rio
 */
 
-char *rcs_mem = "$Id: mem.c,v 1.12 1996/05/06 16:59:00 roberto Exp roberto $";
+char *rcs_luamem = "$Id: luamem.c,v 1.12 1996/05/06 16:59:00 roberto Exp roberto $";
 
 #include <stdlib.h>
 
-#include "mem.h"
+#include "luamem.h"
 #include "lua.h"
 
 
diff --git a/luamem.h b/luamem.h
index cbc973b2..86557802 100644
--- a/luamem.h
+++ b/luamem.h
@@ -1,11 +1,11 @@
 /*
-** mem.c
+** luamem.c
 ** memory manager for lua
-** $Id: mem.h,v 1.7 1996/04/22 18:00:37 roberto Exp roberto $
+** $Id: luamem.h,v 1.7 1996/04/22 18:00:37 roberto Exp roberto $
 */
  
-#ifndef mem_h
-#define mem_h
+#ifndef luamem_h
+#define luamem_h
 
 #ifndef NULL
 #define NULL 0
diff --git a/makefile b/makefile
index cb0b523b..93e5113b 100644
--- a/makefile
+++ b/makefile
@@ -28,7 +28,7 @@ LUAOBJS = \
 	inout.o \
 	tree.o \
 	fallback.o \
-	mem.o \
+	luamem.o \
 	func.o \
 	undump.o
 
@@ -75,24 +75,24 @@ clear	:
 	co $@
 
 
-fallback.o : fallback.c mem.h fallback.h lua.h opcode.h types.h tree.h func.h \
+fallback.o : fallback.c luamem.h fallback.h lua.h opcode.h types.h tree.h func.h \
   table.h 
-func.o : func.c luadebug.h lua.h table.h tree.h types.h opcode.h func.h mem.h 
-hash.o : hash.c mem.h opcode.h lua.h types.h tree.h func.h hash.h table.h 
+func.o : func.c luadebug.h lua.h table.h tree.h types.h opcode.h func.h luamem.h 
+hash.o : hash.c luamem.h opcode.h lua.h types.h tree.h func.h hash.h table.h 
 inout.o : inout.c lex.h opcode.h lua.h types.h tree.h func.h inout.h table.h \
-  mem.h 
+  luamem.h 
 iolib.o : iolib.c lua.h luadebug.h lualib.h 
-lex.o : lex.c mem.h tree.h types.h table.h opcode.h lua.h func.h lex.h inout.h \
+lex.o : lex.c luamem.h tree.h types.h table.h opcode.h lua.h func.h lex.h inout.h \
   luadebug.h parser.h 
 lua.o : lua.c lua.h lualib.h 
 mathlib.o : mathlib.c lualib.h lua.h 
-mem.o : mem.c mem.h lua.h
-opcode.o : opcode.c luadebug.h lua.h mem.h opcode.h types.h tree.h func.h hash.h \
+luamem.o : luamem.c luamem.h lua.h
+opcode.o : opcode.c luadebug.h lua.h luamem.h opcode.h types.h tree.h func.h hash.h \
   inout.h table.h fallback.h undump.h 
-parser.o : parser.c luadebug.h lua.h mem.h lex.h opcode.h types.h tree.h func.h \
+parser.o : parser.c luadebug.h lua.h luamem.h lex.h opcode.h types.h tree.h func.h \
   hash.h inout.h table.h 
 strlib.o : strlib.c lua.h lualib.h 
-table.o : table.c mem.h opcode.h lua.h types.h tree.h func.h hash.h table.h \
+table.o : table.c luamem.h opcode.h lua.h types.h tree.h func.h hash.h table.h \
   inout.h fallback.h luadebug.h 
-tree.o : tree.c mem.h lua.h tree.h types.h lex.h hash.h opcode.h func.h table.h 
-undump.o : undump.c opcode.h lua.h types.h tree.h func.h mem.h table.h undump.h 
+tree.o : tree.c luamem.h lua.h tree.h types.h lex.h hash.h opcode.h func.h table.h 
+undump.o : undump.c opcode.h lua.h types.h tree.h func.h luamem.h table.h undump.h 
diff --git a/opcode.c b/opcode.c
index 107024a4..0550386a 100644
--- a/opcode.c
+++ b/opcode.c
@@ -11,7 +11,7 @@ char *rcs_opcode="$Id: opcode.c,v 3.76 1996/09/24 21:46:44 roberto Exp roberto $
 #include <stdlib.h>
 
 #include "luadebug.h"
-#include "mem.h"
+#include "luamem.h"
 #include "opcode.h"
 #include "hash.h"
 #include "inout.h"
diff --git a/table.c b/table.c
index 8af20ccc..608c4f7a 100644
--- a/table.c
+++ b/table.c
@@ -5,7 +5,7 @@
 
 char *rcs_table="$Id: table.c,v 2.57 1996/07/12 20:00:26 roberto Exp roberto $";
 
-#include "mem.h"
+#include "luamem.h"
 #include "opcode.h"
 #include "tree.h"
 #include "hash.h"
diff --git a/tree.c b/tree.c
index 5ebdf84b..950ba414 100644
--- a/tree.c
+++ b/tree.c
@@ -8,7 +8,7 @@ char *rcs_tree="$Id: tree.c,v 1.19 1996/02/22 20:34:33 roberto Exp $";
 
 #include <string.h>
 
-#include "mem.h"
+#include "luamem.h"
 #include "lua.h"
 #include "tree.h"
 #include "lex.h"
diff --git a/undump.c b/undump.c
index a3eb881f..df6a8390 100644
--- a/undump.c
+++ b/undump.c
@@ -8,7 +8,7 @@ char* rcs_undump="$Id: undump.c,v 1.20 1996/11/16 20:14:23 lhf Exp lhf $";
 #include <stdio.h>
 #include <string.h>
 #include "opcode.h"
-#include "mem.h"
+#include "luamem.h"
 #include "table.h"
 #include "undump.h"
 
